trainee_template/ablation.amlg

```diff
@@ -0,0 +1,173 @@
+(null
+
+    ;compute the entropy of the influence weights of the context's influential cases
+    ; context_values: list of context values to react to
+    ; context_features: list of context features to react to
+    ; use_case_weights: optional flag default true. if false, case weights will not be used during the react
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #ComputeInfluenceWeightEntropyUsingContext
+    (declare
+        (assoc
+            context_values (list)
+            context_features (list)
+            use_case_weights (true)
+            weight_feature ".case_weight"
+        )
+
+        (if (not use_case_weights)
+            (assign weight_feature ".none"))
+
+        ;compute the entropy of the influence weights as retrieved by react
+        (entropy (map
+            (lambda (get (current_value) ".influence_weight"))
+
+            (get
+                (call React (assoc
+                    context_features context_features
+                    context_values context_values
+                    details (assoc
+                        influential_cases (true)
+                    )
+                    use_case_weights use_case_weights
+                    weight_feature weight_feature
+
+                ))
+                "influential_cases"
+            )
+        ))
+    )
+
+    ;compute the entropy of the influence weights of an existing case's influential cases
+    ; case_id: the case id in the model for which to compute influence weight entropy
+    ; context_features: list of context features to react to
+    ; use_case_weights: optional flag default true. if false, case weights will not be used during the react
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #ComputeInfluenceWeightEntropyUsingCaseID
```

Comment:
can these two methods be combined into one since they have a lot of code overlap, with added logic to decide whether to use caseId or context provided which parameter is specified?